### PR TITLE
Add an option to disable meta_index writes

### PR DESCRIPTION
**Pull request by tobim**

### CODE REVIEW

File: nix/vast/default.nix [hunk 1]
```diff
@@ -17,6 +17,7 @@
 , python3
 , jq
 , tcpdump
+, utillinux
```
Comment (mavam) [inline nix/vast/default.nix:20]: What's this?

Comment (dominiklohmann) [reply nix/vast/default.nix:20]: A small fix that enables colored log output in the Nix builds. We should cherry-pick it into master.

Comment (mavam) [reply nix/vast/default.nix:20]: Yes please.

File: libvast/src/system/application.cpp [hunk 2]
```diff
@@ -64,7 +64,9 @@ command::opts_builder add_index_opts(command::opts_builder ob) {
                                             "partitions")
     .add<size_t>("max-taste-partitions", "maximum number of immediately "
                                          "scheduled partitions")
-    .add<size_t>("max-queries,q", "maximum number of concurrent queries");
+    .add<size_t>("max-queries,q", "maximum number of concurrent queries")
+    .add<bool>("disable-recoverability", "don't sync meta-index for every new "
+                                         "partipartition");
```
Comment (lava) [inline libvast/src/system/application.cpp:69]: `s/participation/partition/`

File: libvast/src/system/index.cpp [hunk 3]
```diff
@@ -97,9 +97,9 @@ index_state::~index_state() {
     flush_to_disk();
 }
 
-caf::error
-index_state::init(const path& dir, size_t max_partition_size,
-                  uint32_t in_mem_partitions, uint32_t taste_partitions) {
+caf::error index_state::init(const path& dir, size_t max_partition_size,
+                             uint32_t in_mem_partitions,
+                             uint32_t taste_partitions, bool yolo_mode) {
```
Comment (lava) [inline libvast/src/system/index.cpp:102]: `s/yolo_mode/delay_flush_until_shutdown/`

**Review by lava — APPROVED** [review]
I know this wasnt meant to be committed, but on reviewing this PR I don't really see why not. We can (in principle) recover the index state from the partition data, so delaying the write at the cost of potentially having to manually restore it after unclean shutdowns seems like a trade-off that can be reasonable in some situations.

Especially since the current way is not extremely robust either, when the process gets aborted while writing the index state we end up in the same situation where the index state and the on-disk partitions are out of sync.

File: libvast/src/system/application.cpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/src/system/application.cpp:69]:
```suggestion
                                         "partition");
```

File: libvast/src/system/index.cpp (hunk 3 above)
Comment (dominiklohmann) [reply libvast/src/system/index.cpp:102]:
```suggestion
                             uint32_t taste_partitions,
                             bool delay_flush_until_shutdown) {
```

**Pull request merged**